src/streamlink/plugins/piaulizaportal.py

```diff
@@ -0,0 +1,62 @@
+"""
+$description Japanese live-streaming and video hosting platform owned by PIA Corporation.
+$url ulizaportal.jp
+$type live, vod
+$account Purchased tickets are required.
+$notes Tickets purchased at "PIA LIVE STREAM" are used for this platform.
+"""
+
+import re
+import time
+
+from streamlink.exceptions import FatalPluginError
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+
+@pluginmatcher(
+    re.compile(
+        r"https://ulizaportal\.jp/pages/(?P<id>[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12})(\?expires=(?P<expires>\d+).*)?",
+    ),
+)
+class PIAULIZAPortal(Plugin):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.session.http.headers.update({"Referer": "https://ulizaportal.jp/"})
+
+    def _get_streams(self):
+        self.id = self.match.group("id")
+
+        expires = self.match.group("expires")
+        if expires and int(expires) <= time.time():
+            raise FatalPluginError("The link is expired")
```

Comment:
> Invalid / expired input URLs should just return `None` and log an error message.

So, `--retry-streams` and `--retry-max` will retry expired links...? I think it's in vain.

> `PluginError` and `FatalPluginError` are reserved for other use cases.

OK. The reason I chose `FatalPluginError` is that it can stop the retry loop. 

https://github.com/streamlink/streamlink/blob/54136f414cf7c8f5054990d45844af0ee8dd6064/src/streamlink_cli/main.py#L464-L472

<br>

Changing `FatalPluginError` to error messages does not hurt, though.